released/discovery_center/mission_4033/main.tf

```diff
@@ -0,0 +1,201 @@
+  for_each             = toset("${var.subaccount_admins}")
+  subaccount_id        = btp_subaccount.project.id
+  role_collection_name = "Subaccount Administrator"
+  user_name            = each.value
+}
+
+###############################################################################################
+# Assignment of users as sub account service administrators
+###############################################################################################
+resource "btp_subaccount_role_collection_assignment" "subaccount-service-admins" {
+  for_each             = toset("${var.subaccount_service_admins}")
+  subaccount_id        = btp_subaccount.project.id
+  role_collection_name = "Subaccount Service Administrator"
+  user_name            = each.value
+}
+
+######################################################################
+# Add "sleep" resource for generic purposes
+######################################################################
+resource "time_sleep" "wait_a_few_seconds" {
+  create_duration = "30s"
+}
+
+######################################################################
+# Setup Kyma
+######################################################################
+data "btp_regions" "all" {}
+
+locals {
+  subaccount_iaas_provider = [for region in data.btp_regions.all.values : region if region.region == data.btp_subaccount.this.region][0].iaas_provider
+}
+
+data "btp_subaccount" "this" {
+  id = btp_subaccount.project.id
+}
+
+resource "btp_subaccount_entitlement" "kymaruntime" {
+  subaccount_id = btp_subaccount.project.id
+  service_name = "kymaruntime"
+  plan_name    = lower(local.subaccount_iaas_provider)
+  amount       = 1
+}
+
+
+resource "btp_subaccount_environment_instance" "kyma" {
+  subaccount_id    = btp_subaccount.project.id
+  name             = var.kyma_instance.name
+  environment_type = "kyma"
+  service_name     = "kymaruntime"
+  plan_name        = "aws"
+  parameters = jsonencode({
+    name            = var.kyma_instance.name
+    region          = var.kyma_instance.region
+    machine_type    = var.kyma_instance.machine_type
+    auto_scaler_min = var.kyma_instance.auto_scaler_min
+    auto_scaler_max = var.kyma_instance.auto_scaler_max
+  })
+  timeouts = {
+    create = var.kyma_instance.createtimeout
+    update = var.kyma_instance.updatetimeout
+    delete = var.kyma_instance.deletetimeout
+  }
+  depends_on = [ btp_subaccount_entitlement.kymaruntime ]
+}
+
+# module "sap_kyma_instance" {
+#   source            = "../../../in-development/modules/envinstance-kyma"
+#   subaccount_id     = btp_subaccount.project.id
+#   name              = var.kyma_instance.name
+# }
+
+
+######################################################################
+# Entitlement of all services
+######################################################################
+resource "btp_subaccount_entitlement" "name" {
+  depends_on = [time_sleep.wait_a_few_seconds]
```

Comment:
This depends on can be remove. The entitlement depends on the subaccount and this dependency is implicitly given by referencing the ID